### DISCUSSION TITLE
Welcome to ofxtools Discussions!

### DISCUSSION THREAD

**Discussion by csingley** [csingley/ofxtools#111] [General]
## Nanu Nanu
  I'm setting this up as a space for `ofxtools` users to discuss relevant matters not directly related to a bug ticket or feature request.  I imagine the primary topic will be tips & tricks to get various FIs responding.

**Comment by jimbeauxh** [comment]
Hi, I'm a first time user of ofxtools. Useful stuff. I see that TAX1099RQ is listed as experimental. Is it usable from ofxget? I couldn't find a way to specify the taxyear on the command line. When I try ofxget tax1099 with Fidelity, I get a successful SIGNONMSGSRSV1, but no payload. When I try it with E*Trade, I get a timeout in urlopen after 30 seconds. Thanks, Jim

**Comment by csingley** [comment]
Welcome Jim.

> I see that TAX1099RQ is listed as experimental. Is it usable from ofxget?

"Experimental" here means "completely untested", with an implication of "by all means submit a pull request if you can get it to stop exploding all over your children and valuables".

During tax season last year I implemented the data model for the tax stuff as it existed at the time... ofxtools ought to be able parse TAX1099RS.  This is the core of the library.  While I was at it I went ahead & finished out a method for `ofxget` to use, but too late to actually try to download anything to use in doing my taxes... the FIs have a habit of shutting off that service not too long after tax day.

I am currently of a mind to take full advantage of this year's much-retarded filing deadlines (they're not even demanding that f1120 or f1065 be filed yet, glory be) so I haven't gotten around to fiddling with this yet. It is on my TODO list this year though, but you are in the van here.

Access is highly FI specific.  E*Trade turned off their OFX endpoint some time ago; I can only get STMTRS out of them by downloading from their website.

I don't currently have accounts at Fidelity, but I'm pretty sure they feature prominently in that big list inside of Turbotax.  Can you get a STMTRS from Fidelity?  It's probably the case that I guessed wrong on how they want the TAX1099RQ formatted, and it might only be fixable by packet-sniffing Turbotax traffic.

I don't have answers at this point, only questions.  Let me know what you run into, cheers.

**Comment by jimbeauxh** [comment]
Hi Christopher,

Using ofxget, I'm able to get INVSTMTRS from both Fidelity and E*Trade. 
At the current moment, I'm more interested in TAX1099RQ :) , which I'm 
not getting at all. And the websites don't seem to support direct OFX 
downloads for 1099s. Until this year, I was sniffing the TurboTax cache 
directory for them, but I hit a TurboTax transaction limit, so the files 
don't even make it into the TT cache. Do you know of any financial apps 
out there that can get these 1099 OFXs?

When I have a chance, I'll play around more with ofxtools to see if I 
can get their servers to cough up a little more. I'd rather go direct.

- Jim

**Comment by csingley** [comment]
> Using ofxget, I'm able to get INVSTMTRS from both Fidelity and E*Trade.

Hey you're right, E*Trade is working again - cool!

So if I ask E*Trade to give me the f1099, I get this:

```
OFXHEADER:100
DATA:OFXSGML
VERSION:102
SECURITY:NONE
ENCODING:USASCII
CHARSET:NONE
COMPRESSION:NONE
OLDFILEUID:NONE
NEWFILEUID:F8182095-03D4-4BFF-8519-90498F4248AF

<OFX>
<SIGNONMSGSRSV1>
<SONRS>
<STATUS>
<CODE>0
<SEVERITY>INFO
</STATUS>
<DTSERVER>20210313140622.000
<LANGUAGE>ENG
<DTPROFUP>20021119140000
</SONRS>
</SIGNONMSGSRSV1>
<TAX1099MSGSRSV1>
<TAX1099TRNRS>
<TRNUID>683D4075-69B5-4EDD-9FDF-F2F4748BF860
<STATUS>
<CODE>14501
<SEVERITY>ERROR
<MESSAGE>1099 forms not available for user.
</STATUS>
</TAX1099TRNRS>
</TAX1099MSGSRSV1>
</OFX>
```

I'm not sure if E*Trade supports direct download by TurboTax.  If they do, it might be at a separate URL.  But the basic TAX1099RQ transport appears to be working within `ofxget`.

If you're trying to get this working, you can pass the `--dryrun` argument (e.g. `ofxget tax1099 etrade -n`) to have look at what payload you're sending to the server.  There are plenty of formatting switches in the CLI to experiment with.

I'm not sure what kind of debug logs Turbotax offers... whether they contain the server URL and OFX request data.  Failing that, users have experienced success by setting up a transparent proxy (e.g. `mitmproxy`) and forcing the Turbotax host to use that proxy, in order to snoop the RQ/RS packets.  I haven't yet gone there, but this might be the year!

**Comment by csingley** [comment]
Follow-up:  looks like the E*Trade account I was using doesn't get a 1099 this year; nothing's available for D/L from the website either.

You aren't getting the same results as me?  My config looks like this:

```
> ofxget list etrade

E*TRADE
ofxhome = 446
clientuid = 
url = https://ofx.etrade.com/cgi-ofx/etradeofx
version = 102
org = fldProv_mId
fid = fldProv_mProvBankId
brokerid = etrade.com
investment = [redacted]
user = [redacted]

```
